### ISSUE TITLE
`useFolderTitleFromIndexFile` doesn't get title from `folder/index.md`

### ISSUE THREAD

**Issue by snowykami** [jooy2/vitepress-sidebar#170]
**Describe the bug**
Thank you for this awesome plugin
When I set `useFolderTitleFromIndexFile` to `true`, the option does not take effect and still uses the folder name as the name of the sidebar option. Moreover, both `useTitleFromFileHeading` and `useTitleFromFrontmatter` are set to `true`, and the current folder contains an `index.md` file with a title: `MyTitle` in the frontmatter.

**To Reproduce**
Use the following options
```typescript
let defaultLocale = 'zh';
const commonSidebarOptions = {
    collapsed: true,
    convertSameNameSubFileToGroupIndexPage: true,
    useTitleFromFrontmatter: true,
    useTitleFromFileHeading: true,
    useFolderTitleFromIndexFile: true,
    useFolderLinkFromIndexFile: true,
    includeFolderIndexFile: true,
}
...
```
```typescript
...
sidebar: generateSidebar(
            [
                ...[defaultLocale, 'en', 'ja', 'zht'].map((locale) => {
                    if (locale === defaultLocale) {
                        return {
                            basePath: '/api/',
                            scanStartPath: `${locale}/api`,
                            resolvePath: '/api/',
                            ...commonSidebarOptions
                        }
                    } else {
                        return {
                            basePath: `/${locale}/api/`,
                            scanStartPath: `${locale}/api`,
                            resolvePath: `/${locale}/api/`,
                            ...commonSidebarOptions
                        }
                    }
                })
            ]
        ),
```

**Expected behavior**
The sidebar will display as follows:
```
mbcp.presets V
    mbcp.presets
    mbcp.model
        mbcp.presets.model
```

But it actually displays like this
```
presets
    mbcp.presets
    model
        mbcp.presets.model
```
**Screenshots**
Actual display effect
![image](https://github.com/user-attachments/assets/bb465a5b-6e8f-4e4b-82e5-ef097f545406)

Expected effect
![image](https://github.com/user-attachments/assets/5c3a1ba0-d17f-4af5-b693-e352e680481a)

index.md
![image](https://github.com/user-attachments/assets/bf4e7bac-fe76-4b75-a63a-51ea91004029)
![image](https://github.com/user-attachments/assets/76ebcdb6-b909-49bf-94e1-8ef9eec20b54)

**Desktop**

- OS: Windows 11 26120.1542
- Browser Microsoft Edge
- Node: v20.11.0
- vitepress-sidebar: 1.25.0
- vitepress: 1.3.4

**Additional context**
My instance docs with this issue: [MBCP docs](https://mbcp.sfkm.me/)
Docs repo url: https://github.com/snowykami/mbcp/tree/main/docs

**Comment by snowykami** [comment]
I created a file with the same name as the folder, which met my expectations. However, if there happens to be a file with the same name conflicting with the index, I still need to read the title from the index

**Comment by jooy2** [comment]
Hello! Thank you for using `vitepress-sidebar`.

I have just released package `1.25.3` which fixes this issue.

The `convertSameNameSubFileToGroupIndexPage` option refers to a link in the `.md` file, which is usually the name of the same subfile, and the menu name gets the name of the subfile first. Therefore, if a frontmatter or `h1` heading exists in that file, the name of this file is used first.

What i've fixed now is that if this subfile fails to get a name from the frontmatter and `h1` tags, it will get the title from the index file, depending on whether the option `useFolderTitleFromIndexFile` is enabled. If that option is `false`, it will use the filename (if all else fails).

Could you please upgrade to version `1.25.3` and check if your requirements are reflected?
If the issue no longer occurs, please let me know and i'll move the issue to complete.

Regards,

**Comment by snowykami** [comment]
OK, It is now working according to my expectations, thx~
